sdks/java/core/src/main/java/org/apache/beam/sdk/util/LzoCompressorInputStream.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import io.airlift.compress.lzo.LzoCodec;
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.commons.compress.compressors.CompressorInputStream;
+import org.apache.commons.compress.utils.CountingInputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.apache.commons.compress.utils.InputStreamStatistics;
+
+/**
+ * {@link CompressorInputStream} implementation to create LZO encoded stream. Library relies on <a
+ * href="https://github.com/airlift/aircompressor/">LZO</a>
+ *
+ * @since 1.18
+ */
+public class LzoCompressorInputStream extends CompressorInputStream
```

Comment:
Instead of creating a wrapper class that delegates to lzoIS, create a static method which returns the LZO and LZOP input and output streams and invoke the appropriate static method from the enum readDecompressed/writeDecompressed

Putting the code into a static method will prevent the LzoCodec/LzopCodec from being loaded till the static method is called.

Reply:
Where do you suggest to keep static methods? Currently we have wrapper class in util package.

Reply:
A class called LzoCompression in util is fine.

Reply:
replaced wrapper classes with static methods